finetune/target_models/sequence_labeling.py

```diff
@@ -265,6 +306,45 @@ def predict(self, X, per_token=False, context=None, **kwargs):
             subtoken_predictions=self.config.subtoken_predictions,
         )
 
+        if context is not None and self.config.seq_labeling_2d_spans:
+            doc_annotations_out = []
+            for annotation, sample_context in zip(doc_annotations, context):
+                for single_label in annotation:
+                    single_label["contexts"] = []
+                    for cont in sample_context:
+                        if sequences_overlap(cont, single_label):
+                            single_label["contexts"].append(cont) # cannot break here because one label has multiple contexts.
+                    if not single_label["contexts"] and single_label["text"].strip():
+                        raise FinetuneError("Context does not cover char idx: {} to {}".format(single_label["start"], single_label["end"]))
+                    single_label["start_segments"] = [single_label.pop("start")]
+                    single_label["end_segments"] = [single_label.pop("end")]
+                    single_label["text_segments"] = [single_label.pop("text")]
+                    single_label["confidence_segments"] = [single_label.pop("confidence")]
+      
+                for _ in range(len(annotation)):
```

Comment:
maybe a quick comment that this section combines annotations based on 2d position?